lib/recode/task/remove_parens.ex

```diff
@@ -0,0 +1,89 @@
+defmodule Recode.Task.RemoveParens do
+  @shortdoc "Removes parens from locals without parens"
+
+  @moduledoc """
+  Don't use parens for functions that don't need them.
+
+          # preferred
+          assert true == true
+
+          # not preferred
+          assert(true == true)
+
+  This task rewrites the code when `mix recode` runs with `autocorrect: true`.
+  """
+
+  use Recode.Task, corrector: true, category: :readability
+
+  alias Recode.Issue
+  alias Recode.Task.RemoveParens
+  alias Rewrite.Source
+  alias Sourceror.Zipper
+
+  @impl Recode.Task
+  def run(source, opts) do
+    formatter_opts = Mix.Tasks.Format.formatter_opts_for_file(".mix.exs")
+    locals_without_parens = Keyword.get(formatter_opts, :locals_without_parens, [])
+
+    {zipper, issues} =
+      source
+      |> Source.get(:quoted)
+      |> Zipper.zip()
+      |> Zipper.traverse([], fn zipper, issues ->
+        remove_parens(locals_without_parens, zipper, issues, opts[:autocorrect])
+      end)
+
+    case opts[:autocorrect] do
+      true ->
+        Source.update(source, RemoveParens, :quoted, Zipper.root(zipper))
+
+      false ->
+        Source.add_issues(source, issues)
+    end
+  end
+
+  defp remove_parens(
+         locals_without_parens,
+         %Zipper{node: {fun, _, args} = node} = zipper,
+         issues,
+         true
+       ) do
+    node =
+      Enum.reduce(locals_without_parens, node, fn
+        {^fun, arity}, node when length(args) == arity ->
+          {fun, meta, args} = node
+          meta = Keyword.delete(meta, :closing)
+
+          {fun, meta, args}
+
+        _, node ->
+          node
+      end)
+
+    {%Zipper{zipper | node: node}, issues}
+  end
```

Comment:
What do you think about adding a function `local_witout_parent?/3`:
```elixir
  defp local_without_parens?(locals_without_parens, fun, [_ | _] = args) do
    arity = length(args)

    Enum.any?(locals_without_parens, fn
      {^fun, :*} -> true
      {^fun, ^arity} -> true
      _other -> false
    end)
  end

  defp local_without_parens?(_locals_without_parens, _fun, _args), do: false

  defp remove_parens(
         locals_without_parens,
         %Zipper{node: {fun, meta, args}} = zipper,
         issues,
         true
       ) do
    if local_without_parens?(locals_without_parens, fun, args) do
      node = {fun, Keyword.delete(meta, :closing), args}
      {%Zipper{zipper | node: node}, issues}
    else
      {zipper, issues}
    end
  end
```
This allows us to simplify `remove_parens/4` and observe the value `:*` in `locals_without_parens`.

Reply:
LGTM!  Yes sorry, I should have marked this as a draft.  The duplication was just to get it working and I'm happy to use this code!  I did indeed observe `:*` and will have to look that up, though :)